src/merchants/stripe/StripeMerchant.php

```diff
@@ -173,8 +174,31 @@ public function completePurchase($data)
                 ->setTime(new DateTime());
         }
 
-        if (isset($response->getData()['error']['message'])) {
-            throw new MerchantException('Failed to charge card: ' . $response->getData()['error']['message']);
+        if (
+            isset($response->getData()['last_payment_error'])
+            && $response->getData()['last_payment_error']['code'] === 'card_declined'
+            && $response->getData()['last_payment_error']['decline_code'] === 'insufficient_funds'
+        ) {
+            return (new InsufficientFundsResponse())
+                ->setIsSuccessful(true)
```

Comment:
why successful for failed transaction?
```suggestion
                ->setIsSuccessful(false)
```